test/test_log.cpp

```diff
@@ -54,13 +54,19 @@ BOOST_AUTO_TEST_CASE( pointer_yields_its_value_when_serialized )
 {
     {
         int i = 0;
+        std::ostringstream s;
+        s << &i;
+        const std::string pointerValue = s.str();
```

Comment:
nit: could probably be `const auto`?

Reply:
While I do like `auto` I tend to use it only when the type appears on the right hand side. E.g. `auto p = static_cast<int>(u)`. I find that more readable in the end. This case could be considered a corner case as `str()` is almost `string` but I think it is still better to explicitly see that `pointerValue` is a string.